### PR TITLE
Add `flaky_test` decorator and use it to re-try `test_affine_widget`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Related to #5334. This is a workaround to at least re-try any test using a new decorator. 

This PR apply this to `test_affine_widget`. This one is a bit of a nuisance.

Sample output using ```pytest tests/core -k test_some_flaky_test``` with this example test:
``` python
from tests.conftest import flaky_test


@flaky_test(times=3)
def test_some_flaky_test():
    assert 42 == 24

```
output:
``` python
-------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
FLAKY TEST FAILED (Attempt 1 of 3) - test_utilities::test_some_flaky_test - AssertionError, retrying...
FLAKY TEST FAILED (Attempt 2 of 3) - test_utilities::test_some_flaky_test - AssertionError, retrying...
========================================================================= short test summary info =========================================================================
FAILED tests/core/test_utilities.py::test_some_flaky_test - assert 42 == 24
=================================================================== 1 failed, 5395 deselected in 1.53s ====================================================================
```

### CODE REVIEW

**Comment by user27182** [comment]
Example output from CI:
https://github.com/pyvista/pyvista/actions/runs/10495349023/job/29073616403?pr=6554#step:12:6016
``` python
=================================== FAILURES ===================================
_____________________________ test_some_flaky_test _____________________________

    @flaky_test(times=3)
    def test_some_flaky_test():
>       assert 42 == 24
E       assert 42 == 24

tests/core/test_utilities.py:49: AssertionError
----------------------------- Captured stdout call -----------------------------
FLAKY TEST FAILED (Attempt 1 of 3) - test_utilities::test_some_flaky_test - AssertionError, retrying...
FLAKY TEST FAILED (Attempt 2 of 3) - test_utilities::test_some_flaky_test - AssertionError, retrying...
```

File: tests/plotting/test_widgets.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from pyvista.plotting.affine_widget import DARK_YELLOW
 from pyvista.plotting.affine_widget import get_angle
 from pyvista.plotting.affine_widget import ray_plane_intersection
+from tests.conftest import flaky_test
```
Comment (user27182) [inline tests/plotting/test_widgets.py:13]: Not sure if the decorator should be defined in `conftest`, or elsewhere...

**Comment by user27182** [comment]
The decorator from this PR provides similar functionality to `pytest-repeat` suggested here: https://github.com/pyvista/pyvista/issues/5334#issuecomment-1853500900

File: tests/plotting/test_widgets.py (hunk 1 above)
Comment (tkoyama010) [reply tests/plotting/test_widgets.py:13]: Fine to define in `conftest` .

**Review by tkoyama010 — APPROVED** [review]
OK. Let's try how it works.

**Pull request merged**